src/python/pants/help/help_info_extracter.py

```diff
@@ -497,8 +499,9 @@ def load() -> OptionScopeHelpInfo:
                     )
                 provider = ""
                 if subsystem_cls is not None and build_configuration is not None:
-                    provider = cls.get_provider(
```

Comment:
It looks like there's 7 callers of `get_provider` in this file, so 5 that remain. Did you consider whether those other ones could switch over? And, if so, if we could just replace the implementation of `get_provider` rather than adding `get_closest_provider_for_subsystem`?

Reply:
The issue is that not all of those calls have a subsystem class available; I can only use the new function where a subsystem class is present to compare against. I'll still look through the rest though.

Reply:
Ah, of course. I guess some might have something that's not a subsystem class that could be used as an "anchor", but maybe not all of them.

Reply:
I can confirm that of the remaining 5 callsites for `get_provider`, none of them have a subsystem class available. So we can't change those without adding some more methods for target types and union rules to return a better provider choice.